jsonrpc.py

```diff
@@ -0,0 +1,149 @@
+# SPDX-License-Identifier: GPL-3.0
+import io
+import json
+import os
+import re
+
+import xbmc
+
+
+class JsonRpcApi(object):
+    def __init__(self):
+        """ Initialise the JSON RPC API Implementation. """
+        self._paths = xbmc.get_add_on_info_from_calling_script()
```

Comment:
Now each and every RPC call will try to execute this, which is not really needed. I would say that we should pass the path from `executeJSONRPC` instead of calling it here. If you feel that it makes more sense in this file, only load the paths when reading the settings.xml in the `__init__` of the `Settings` class.

And with that, it makes little sense to have a `JsonRpcApi` class at all.

Reply:
You are right that I should probably also have cached it like at the bottom in `xbmc.py`. I initially wanted to use the `xbmc.__add_on_info` var directly that already contains the data I need, but the naming prevented that.